destination.go

```diff
@@ -68,20 +74,39 @@ func (d *Destination) Configure(ctx context.Context, cfg map[string]string) erro
 	if err != nil {
 		return fmt.Errorf("invalid config: %w", err)
 	}
+	if !d.config.TLSDisable {
+		d.clientCert, d.caCertPool, err = d.config.ParseMTLSFiles()
+		if err != nil {
+			return fmt.Errorf("invalid mTLS config: %w", err)
+		}
+	}
 	return nil
 }
 
 func (d *Destination) Open(ctx context.Context) error {
 	dialOptions := []grpc.DialOption{
 		grpc.WithContextDialer(d.dialer),
-		grpc.WithInsecure(), //nolint:staticcheck // todo: will use mTLS with connection
+		grpc.WithTransportCredentials(insecure.NewCredentials()), // todo: will use mTLS with connection
```

Comment:
Looks like we don't need this option here anymore.

Reply:
oh I missed this up while solving the conflicts to change the PR base, but should be fine in the server PR